be1-go/popcha/resources/popcha.html

```diff
@@ -0,0 +1,88 @@
+
+<!DOCTYPE html>
+<html lang="en-us">
+<script>
+    // on window loading, create websocket connection
+    window.addEventListener("load", (_) => {
+        const socket = new WebSocket("{{.WebSocketAddr}}");
+        socket.onopen = (_) => {
+            console.log("Starting webSocket connection with address: "+ "{{.WebSocketAddr}}");
+            console.log("redirect URI: " + "{{.RedirectHost}}");
+        }
+        // when receiving the parameters, construct the redirect URI
+        socket.onmessage = (event) => {
+            let redir = "{{.RedirectHost}}";
+            if (redir.startsWith("localhost")) {
+                redir = "http://" + redir
+            }
```

Comment:
It might  be good to remove this later (since https is part of our threat model's assumptions), but it is ok to have it for now.

Reply:
Or mkcert might be a solution: [](https://github.com/FiloSottile/mkcert/releases)

Reply:
I will look into it, thanks